service/rpc2/server.go

```diff
@@ -0,0 +1,466 @@
+}
+
+type StacktraceIn struct {
+	Id    int
+	Depth int
+	Full  bool
+}
+
+func (s *RPCServer) Stacktrace(arg StacktraceIn, locations *[]api.Stackframe) error {
+	locs, err := s.debugger.Stacktrace(arg.Id, arg.Depth, arg.Full)
+	if err != nil {
+		return err
+	}
+	*locations = locs
+	return nil
+}
+
+type ListBreakpointsIn struct {
+}
+
+func (s *RPCServer) ListBreakpoints(arg ListBreakpointsIn, breakpoints *[]*api.Breakpoint) error {
+	*breakpoints = s.debugger.Breakpoints()
+	return nil
+}
+
+func (s *RPCServer) CreateBreakpoint(bp, newBreakpoint *api.Breakpoint) error {
+	createdbp, err := s.debugger.CreateBreakpoint(bp)
+	if err != nil {
+		return err
+	}
+	*newBreakpoint = *createdbp
+	return nil
+}
+
+type ClearBreakpointIn struct {
+	Id   int
+	Name string
+}
+
+func (s *RPCServer) ClearBreakpoint(arg ClearBreakpointIn, breakpoint *api.Breakpoint) error {
+	var bp *api.Breakpoint
+	if arg.Name != "" {
+		bp = s.debugger.FindBreakpointByName(arg.Name)
+		if bp == nil {
+			return fmt.Errorf("no breakpoint with name %s", arg.Name)
+		}
+	} else {
+		bp = s.debugger.FindBreakpoint(arg.Id)
+		if bp == nil {
+			return fmt.Errorf("no breakpoint with id %d", arg.Id)
+		}
+	}
+	deleted, err := s.debugger.ClearBreakpoint(bp)
+	if err != nil {
+		return err
+	}
+	*breakpoint = *deleted
+	return nil
+}
+
+type AmendBreakpointOut struct {
+}
+
+func (s *RPCServer) AmendBreakpoint(amend *api.Breakpoint, out *AmendBreakpointOut) error {
+	return s.debugger.AmendBreakpoint(amend)
+}
+
+type ListThreadsIn struct {
+}
+
+func (s *RPCServer) ListThreads(arg ListThreadsIn, threads *[]*api.Thread) (err error) {
+	*threads, err = s.debugger.Threads()
+	return err
+}
+
+type GetThreadIn struct {
+	Id int
+}
+
+func (s *RPCServer) GetThread(arg GetThreadIn, thread *api.Thread) error {
+	t, err := s.debugger.FindThread(arg.Id)
+	if err != nil {
+		return err
+	}
+	if t == nil {
+		return fmt.Errorf("no thread with id %d", arg.Id)
+	}
+	*thread = *t
+	return nil
+}
+
+type ListPackageVarsIn struct {
+	Filter string
+}
+
+func (s *RPCServer) ListPackageVars(arg ListPackageVarsIn, variables *[]api.Variable) error {
+	state, err := s.debugger.State()
+	if err != nil {
+		return err
+	}
+
+	current := state.CurrentThread
+	if current == nil {
+		return fmt.Errorf("no current thread")
+	}
+
+	vars, err := s.debugger.PackageVariables(current.ID, arg.Filter)
+	if err != nil {
+		return err
+	}
+	*variables = vars
+	return nil
+}
+
+type ListRegistersIn struct {
+}
+
+type ListRegistersOut struct {
+	Registers string
+}
+
+func (s *RPCServer) ListRegisters(arg ListRegistersIn, out *ListRegistersOut) error {
+	state, err := s.debugger.State()
+	if err != nil {
+		return err
+	}
+
+	regs, err := s.debugger.Registers(state.CurrentThread.ID)
+	if err != nil {
+		return err
+	}
+	out.Registers = regs
+	return nil
+}
+
+type ListLocalVarsIn struct {
+	Scope api.EvalScope
+}
+
+func (s *RPCServer) ListLocalVars(arg ListLocalVarsIn, variables *[]api.Variable) error {
```

Comment:
I think for the sake of consistency we should also wrap this return value in a struct as well.

```go
struct ListLocalVarsOut {
        variables *[]api.Variable
}
```